### DISCUSSION TITLE
"Discard input?" - more specific answer than "cancel"? (PR opened)

### DISCUSSION THREAD

**Discussion by matkoniecz** [streetcomplete/StreetComplete#3396] [Q&A] [answered]
There is a "Discard input?" menu appearing after user fills answer, at least partially and tries to leave answer window.

It has options "cancel" and "discard" with "cancel" being not entirely clear - as user is cancelling filling of the quest. And "cancel cancelling" is tricky.

On the other hand giving special text here would add even more things to translate...

Still, I think that it would be worth doing. Situation is even worse in some translations where even the best found text is more confusing than English version.

See #3576 for PR.

**Comment by smichel17** [comment]
_Keep_ is the best verb I came up with so far. _Back_ and _stay_ are two other options.

**Reply by matkoniecz** [reply]
Yes, I also thought about "Keep" and it seems an improvement even for English.

**Comment by smichel17** [comment]
Aside: I cannot reproduce it now, but I thought StreetComplete sometimes shows me this dialog even when I have not entered anything (but only after I have interacted with the form in some way). I'll keep an eye out if it happens again.

**Reply by matkoniecz** [reply]
"even when I have not entered anything" in such case it would a bug (in the same way as entering something and leaving form always should show something)

Please report an issue in case of finding something specific, as without that info there is no chance to fix it.

**Comment by westnordost** [comment]
> Still, I think that it would be worth doing. Situation is even worse in some translations where even the best found text is more confusing than English version.

Like French, where "cancel" and "discard" is the same word. But what they did is to use a different word, AFAIR.

**Comment by westnordost** [comment]
In general, I don't see a problem with it. It follows the (Android) guidelines that for things like

"Do thing?" -> "ok" "cancel",

instead

"Do thing?" -> "thing" "cancel",

is displayed. So, Android users should be familiar with these kind of dialogs. I fear that if we would switch out "cancel" with "back", those users might get confused. You could equally argue - back to where? Back to the main map view, closing the quest form? Back to the form being not-filled-in?

**Reply by matkoniecz** [reply]
"back" would be unclear, but what about

> Discard input?
> (Keep) (Discard)

**Reply by westnordost** [reply]
That sounds a bit like it changes are "kept", i.e. applied

**Comment by mnalis** [comment]
> There is a "Discard input?" menu appearing after user fills answer, at least partially and tries to leave answer window.

How about simple `yes` / `no` dialog as an answer to that `Discard input?` question?

It makes sense to me and seems to be very understable (both in English and in Croatian) - even if it does not describe exactly what would happen after you press the button, there would be no confusion what is being asked, and what you need to answer.

**Comment by smichel17** [comment]
> How about simple `yes` / `no` dialog as an answer to that `Discard input?` question?

The Android guidelines suggest using the verb as the action. While guidelines don't always need to be followed, I think this one is good.

However, I also think you're on the right track: `No` seems better than any alternative so far. `Discard Input?` → `No` / `Discard`

**Reply by westnordost** [reply]
I think Discard / no would be ok.

**Reply by matkoniecz** [reply]
Seemed trivial to implement but there are following options:

- change content of `confirmation_discard_negative` from "Cancel" to "No" (many languages translated already 'Cancel')
- use `quest_generic_hasFeature_no` (in some languages the same text may not fitting for both)
- change to `confirmation_discard_negative2` with value "No" (forces translating again)

I think that last solution would be the best.

**Reply by mnalis** [reply]
I also prefer the last option, as it gives the translator most choice (although 2nd option would also work for Croatian in this case). 
But that is pretty visible translation (not one quest only), so if the translators are not quick to translate it (before next release) that might be problematic.  (I don't know the translation dynamics of other languages, so this might not be a problem)

**Reply by matkoniecz** [reply, marked as answer]
I opened https://github.com/streetcomplete/StreetComplete/pull/3576 (as it affects translations)
